distrobuilder/main.go

```diff
@@ -443,18 +443,109 @@ func getDefinition(fname string, options []string) (*shared.Definition, error) {
 	return &def, nil
 }
 
-func fixCapabilities() {
+// addSystemdGenerator creates a systemd-generator which runs on boot, and does some configuration around the system itself and networking.
+func addSystemdGenerator() {
 	// Check if container has systemd
 	if !lxd.PathExists("/etc/systemd") {
 		return
 	}
 
-	os.MkdirAll("/etc/systemd/system/service.d", 0755)
+	content := `#!/bin/sh
 
-	content := `[Service]
-ProtectProc=default
-ProtectControlGroups=no
-ProtectKernelTunables=no
+# NOTE: systemctl is not available for systemd-generators
```

Comment:
We should do a very early check that we are in fact in a lxc container or a LXD VM and exit 0 if not to avoid affecting any other system.